guides/relationships/features/links-mode.md

```diff
@@ -0,0 +1,199 @@
+adopted from the [JSON:API](https://jsonapi.org) specification.
+
+- **meta** *(optional)*: an object of arbitrary extra information about the relationship
+- **links** *(optional)*: an object containing various links for fetching and managing the relationship
+- **data**: the membership (state of) the relationship
+
+This is roughly described by the interface below:
+
+```ts
+interface Relationship {
+  meta?: Record<string, Value>;
+  links?: Links;
+}
+
+interface ResourceRelationship extends Relationship {
+  data: { type: string; id: string | null; lid: string } | null;
+}
+
+interface CollectionRelationship extends Relationship {
+  data: { type: string; id: string | null; lid: string }[]
+}
+```
+
+When `linksMode` is activated for a relationship, it is required that a related link is present.
+
+```ts
+interface LinksModeRelationship {
+  meta?: Record<string, Value>;
+
+  // no longer optional
+  links: {
+    related: string | { href: string };
+
+    // other links still as desired
+  }
+}
+```
+
+### Related Links May be provided by Handlers
+
+This means that in order to use links mode a relationship payload given to the cache MUST contain this related link. If your API does not provide this link, a request handler could be utilized to decorate an API response to add them provided that your handlers (or your API) are able to understand that link.
+
+### When A Relationship is Fetched, the related link is used
+
+Fetching a relationship via any of `relationship.reload`, `reference.reload`, `reference.load` or `await record.relationship` will issue a request to your handler chain. That request will look like the following:
+
+```ts
+interface FetchRelationshipRequest {
+  op: 'findHasMany' | 'findBelongsTo';
+  store: Store;
+  url: string; // the related link
+  method: 'GET';
+  records: StableRecordIdentifier | StableRecordIdentifier[] | null; // the current membership of the relationship
+  data: {
+    useLink: true;
+    field: LegacyBelongsToField | LegacyHasManyField;
+    links: Links;
+    meta: Meta;
+    options: unknown; // any options passed to `reload` or `load`
+    record: StableRecordIdentifier; // the parent record
+  };
+
+  // tells the store not to automatically convert the response into something reactive
+  // since the reactive relationship class itself will do that
+  [EnableHydration]: false; 
+}
+```
+
+The most important 3 things in this request are
+
+- the `op` code: this is how the cache will know to use the response to update the state of a relationship
+- `data.field`: this is how the cache will know which field it should update
+- `data.record`: this is how the cache will know which record to associate the response to.
+
+The normalized API response (what your handler must return either directly from your API or with some normalization on the client) that should be passed to the JSON:API cache should be a standard JSON:API document.
```

Comment:
This is a little confusing bc it makes it sound like I need a custom handler, but I don't think that's true since if the API returns the correct format we can just use the built-in Fetch handler, right?

Reply:
yes